### PR TITLE
#3 CI, Docker Publishing, and Pull Request Template for BigTent

### PR DESCRIPTION
**Pull request by bwmcadams**
### 💻 Description of Change(s) (w/ context)
This introduces CI & Docker Publish for goatrodeo.
- the "Rust Build & Test" job runs `cargo test` and verifies it builds and tests ok against any push in any branch, as well as new pull requests.
- the "Publish Container Images" will build container images and then push them to both Docker Hub & GHCR, with attestations
### 🧠 Rationale Behind Change(s)
We need CI and Docker container publishing.
### 📝 Test Plan
I've tested heavily; the "will it trigger on main when build and test runs" test can only be tested from a merge to main
### 📜 Documentation
There's a `README.md` in the workflow with some introductory data on the new tests
### 💣 Quality Control
(All items must be checked before a PR is merged)
Did you…
- [x] Mention an issue number in the PR title?
- [x] Update the version # in the build file? **this was not needed / appropriate for changing just workflows for actions**
- [x] Create new and/or update relevant existing tests? **this was not needed / appropriate for changing just workflows for actions**
- [x] Create or update relevant documentation and/or diagrams?
- [x] Comment your code?
- [x] Fix any stray verbose logging (removing, or moving to debug / trace level)?

### Before Merging…
- Make sure the Quality Control boxes are all ticked
- Make sure any open comments or conversations on the PR are resolved

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Github Actions Workflows
+
+This folder contains Github Actions workflows
+
+- `rust-ci.yml` uses sbt to run a build & test job
```
Comment (dpp) [inline .github/workflows/README.md:5]: sbt? Not cargo?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM messense/rust-musl-cross:x86_64-musl AS build
+
+LABEL MAINTAINER="engineering@spicelabs.io"
```
Comment (dpp) [inline Dockerfile:3]: Have you set up this alias?

**Review by dpp — APPROVED** [review]
:+1:

File: .github/workflows/README.md (hunk 1 above)
Comment (bwmcadams) [reply .github/workflows/README.md]: fixed.

File: Dockerfile (hunk 2 above)
Comment (bwmcadams) [reply Dockerfile:3]: fixing for ext-engineering instead as discussed.

**Review by dpp — APPROVED** [review]
Awesome 🐩

**Pull request merged**